src/server/game/Entities/Player/Player.cpp

```diff
@@ -2018,14 +2018,19 @@ bool Player::BuildEnumData(PreparedQueryResult result, WorldPacket* data)
     uint8 hairColor = fields[8].GetUInt8();
     uint8 facialStyle = fields[9].GetUInt8();
 
+    uint32 charFlags = 0;
+    uint32 playerFlags = fields[17].GetUInt32();
+    uint16 atLoginFlags = fields[18].GetUInt16();
+    uint32 zone = uint32((atLoginFlags & AT_LOGIN_FIRST) != 0 ? 0 : fields[11].GetUInt16()); // if first login do not show the zone
+
     *data << uint8(skin);
     *data << uint8(face);
     *data << uint8(hairStyle);
     *data << uint8(hairColor);
     *data << uint8(facialStyle);
 
     *data << uint8(fields[10].GetUInt8());                   // level
-    *data << uint32(fields[11].GetUInt16());                 // zone
+    *data << zone;                                           // zone
```

Comment:
Shouldn't this be `*data << uint32(zone);`?

Just referencing to the lines above that have `*data << uint8(skin);` for example

Reply:
@Helias

Reply:
`zone` variable is already casted to `uint32()` at line 2024 at its initialization, so, I don't think it's needed to re-cast it.

probably you already understood but I want to be clear:

actually the zone variable contains `uint32((atLoginFlags & AT_LOGIN_FIRST) != 0 ? 0 : fields[11].GetUInt16())` so, it's like
```C++
*data << uint32((atLoginFlags & AT_LOGIN_FIRST) != 0 ? 0 : fields[11].GetUInt16());  // zone
```
if I do another cast uint32() it will be
```C++
*data << uint32(uint32((atLoginFlags & AT_LOGIN_FIRST) != 0 ? 0 : fields[11].GetUInt16()));
``
so like `uint32(uint32( ...something... )`;

Reply:
Yea but so is `    uint8 skin = fields[5].GetUInt8();` etc...
Perhaps it could be nice to have a standard to go after?

Reply:
mhm, you mean to move the casting next-to *data instead of the initialization?
from the readability yes, but from the evaluation of the expressions it's better to do it in the initialization imho

because it's like:
```C++
int x = 'a';
....
*data << int(x);
```

because the 'a' can be converted into an integer, but it should do when you assign the value, not later. otherwise, the compiler will do it automatically or throw an error or just consider it differently (I think).

anyway, I am not an expert of C++, but in my mind the casting should be at the initialization.

It's nice to have a standard and I can change it, let me think about it, this evening maybe I will update it

Reply:
I just copied the code from the file :D

Reply:
done